src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -137,6 +122,51 @@ private void raiseSQLExceptionIfConnectionIsClosed() throws SQLException {
     }
   }
 
+  /**
+   * Function to send in-band telemetry data about DatabaseMetadata get API calls and their
+   * associated SHOW commands
+   *
+   * @param resultSet The ResultSet generated from the SHOW command in the function call. Can be of
+   *     type SnowflakeResultSet or SnowflakeDatabaseMetaDataResultSet
+   * @param functionName name of DatabaseMetadata API function call
+   * @param catalog database
+   * @param schema schema
+   * @param generalNamePattern name of table, function, etc
+   * @param specificNamePattern name of table column, function parameter name, etc
+   */
+  private void sendInBandTelemetryMetadataMetrics(
+      ResultSet resultSet,
+      String functionName,
+      String catalog,
+      String schema,
+      String generalNamePattern,
+      String specificNamePattern) {
+    String queryId = "";
+    if (resultSet instanceof SnowflakeResultSet) {
```

Comment:
Since it's faster to use instanceof than try/catch, I am now using instanceof to determine which getQueryID() function to use. However, the unwrap() function throws an exception so the try/catches still have to be there since I don't think it's necessary to throw an exception when the queryID cannot be found. An exception should never be thrown in these cases anyway.